### ISSUE TITLE
Cannot create an icosahedral mesh without providing a mesh argument

### ISSUE THREAD

**Issue by xylar** [dengwirda/jigsaw-python#13]
It seems the intention is that one can create an icosahedral mesh without passing a `mesh` object:
https://github.com/dengwirda/jigsaw-python/blob/b6e455c22696568f90a7428a49e03c980892c923/jigsawpy/__init__.py#L105
but when I try this:
```
        jigsawpy.cmd.icosahedron(opts, subdivisions)
```
I get:
```
Traceback (most recent call last):
  File "/home/xylar/miniconda3/envs/compass_jigsaw_mesh/bin/compass", line 33, in <module>
    sys.exit(load_entry_point('compass', 'console_scripts', 'compass')())
  File "/home/xylar/code/compass/separate_mesh_steps/compass/__main__.py", line 63, in main
    commands[args.command]()
  File "/home/xylar/code/compass/separate_mesh_steps/compass/run/serial.py", line 281, in main
    run_step()
  File "/home/xylar/code/compass/separate_mesh_steps/compass/run/serial.py", line 249, in run_step
    test_case.run()
  File "/home/xylar/code/compass/separate_mesh_steps/compass/ocean/tests/global_ocean/mesh/__init__.py", line 109, in run
    super().run()
  File "/home/xylar/code/compass/separate_mesh_steps/compass/testcase.py", line 190, in run
    self._run_step(step, self.new_step_log_file)
  File "/home/xylar/code/compass/separate_mesh_steps/compass/testcase.py", line 306, in _run_step
    step.run()
  File "/home/xylar/code/compass/separate_mesh_steps/compass/mesh/spherical.py", line 350, in run
    self.make_jigsaw_mesh(subdivisions)
  File "/home/xylar/code/compass/separate_mesh_steps/compass/mesh/spherical.py", line 376, in make_jigsaw_mesh
    jigsawpy.cmd.icosahedron(opts, subdivisions)
  File "/home/xylar/miniconda3/envs/compass_jigsaw_mesh/lib/python3.9/site-packages/jigsawpy/__init__.py", line 106, in icosahedron
    return jigsaw.icosahedron(
  File "/home/xylar/miniconda3/envs/compass_jigsaw_mesh/lib/python3.9/site-packages/jigsawpy/jigsaw.py", line 597, in icosahedron
    raise Exception("Incorrect type: MESH.")
Exception: Incorrect type: MESH.
```
It seems that this check:
https://github.com/dengwirda/jigsaw-python/blob/b6e455c22696568f90a7428a49e03c980892c923/jigsawpy/jigsaw.py#L519-L520
should check for `None`:
```
    if (mesh is not None and not isinstance(mesh, jigsaw_msh_t)):
        raise Exception("Incorrect type: MESH.")
```

**Comment by dengwirda** [comment]
Okay yes, I think you're right @xylar --- the check for `None` on `mesh` should be included there. 
In terms of _geometry_ objects though, I believe `opts.geom_file` may also need to point to an `ellipsoid-mesh` geometry definition, as this is where the `radii` information will be read from. Different to the bug re: `mesh=None` though...

**Comment by xylar** [comment]
Sorry for the confusion.  I didn't mean anything to do with the _geometry_ object.

**Comment by dengwirda** [comment]
This should be fixed in #17.
